chainlink-modsec/libmodsec/internal/attestation-indexer/in_memory_indexer.go

```diff
@@ -0,0 +1,125 @@
+package attestation_indexer
+
+import (
+	"context"
+	"fmt"
+	"slices"
+
+	"golang.org/x/sync/syncmap"
+
+	"github.com/smartcontractkit/chainlink-modsec/libmodsec/pkg/modsectypes"
+)
+
+// InMemoryAttestationIndexer is an in-memory implementation of an attestation indexer
+// It stores all proofs in an in-memory map, indexed by messageID([32]byte) and then by veriiferType(enum) and then verifierID([32]byte)
+// This implementation is useful when assuming that attestations will be requested via messageID
+// we only organize by verifierType here for the sake of "VerifierTypeMeetsQuorum" function
+// Normally, we can just get verifierType from verifierID, but that would require loading all verifiers and iterating over them
+// There are significantly more optimizations we can make in a real indexer solution
+/**
+at.attestations: {
+	[messageID1]: {
+		[verifierType1]: {
+			[verifierID1]: proof1,
+			[verifierID2]: proof2,
+		},
+		[verifierType2]: {
+			[verifierID3]: proof3,
+			[verifierID4]: proof4,
+		},
+	},
+}
+*/
+type InMemoryAttestationIndexer struct {
+	attestations syncmap.Map // concurrency-safe map of messageID to map of verifierID to attestation(proof)
+}
+
+// NewInMemoryAttestationIndexer creates a new AttestationIndexer instance
+func NewInMemoryAttestationIndexer() *InMemoryAttestationIndexer {
+	// Create a new InMemoryAttestationIndexer with an initialized attestations map
+	a := &InMemoryAttestationIndexer{
+		attestations: syncmap.Map{},
+	}
+	return a
+}
+
+// GetAttestationsForMessage retrieves attestations for a given message and converts it into the modsectypes.Attestation type
+func (at *InMemoryAttestationIndexer) GetAttestationsForMessage(_ context.Context, msg modsectypes.Message) ([]modsectypes.Attestation, error) {
```

Comment:
Does this method only need message ID? Would make it more explicit if we just take `messageID [32]byte` as an argument rather than the full message.